packages/test-runner/src/fixtures.ts

```diff
@@ -74,15 +74,23 @@ class Fixture {
       return;
     for (const name of this.deps) {
       await this.pool.setupFixture(name, config, info);
-      this.pool.instances.get(name).usages.add(this.name);
+      for (const instance of this.pool.instances.get(name))
+        instance.usages.add(this.name);
     }
 
     const params = {};
-    for (const n of this.deps)
-      params[n] = this.pool.instances.get(n).value;
-    let setupFenceFulfill: { (): void; (value?: unknown): void; };
-    let setupFenceReject: { (arg0: any): any; (reason?: any): void; };
-    const setupFence = new Promise((f, r) => { setupFenceFulfill = f; setupFenceReject = r; });
+    for (const n of this.deps) {
+      if (!this.pool.instancesRan.has(n))
+        this.pool.instancesRan.set(n, 0);
+      else
+        this.pool.instancesRan.set(n, this.pool.instancesRan.get(n) + 1);
+      const fixture = this.pool.instances.get(n)[this.pool.instancesRan.get(n)];
+      params[n] = fixture.value;
+    }
+
+    let setupFenceFulfill: {(): void; (value?: unknown): void;};
```

Comment:
Auto formatting, is hard, I tried my best! We need somehow configure ESLint for that.